common/lib/xmodule/xmodule/modulestore/mixed.py

```diff
@@ -266,6 +266,21 @@ def get_items(self, course_key, **kwargs):
         return store.get_items(course_key, **kwargs)
 
     @strip_key
+    def get_course_summaries(self, **kwargs):
+        """
+        Returns a list containing the course information in CourseSummary objects.
+        Information contains `location`, `display_name`, `locator` of the courses in this modulestore.
+        """
+        course_summaries = {}
+        for store in self.modulestores:
+            for course_summary in store.get_course_summaries(**kwargs):
+                course_id = self._clean_locator_for_mapping(locator=course_summary.id)
+                if course_id not in course_summaries:
+                    # course is indeed unique. save it in result
```

Comment:
Since the XML modulestore no longer stores courses, we should no longer have this case today (since Split and old Mongo have different course_id naming schemes).  

However, we may have additional modulestores in the future (but hopefully not) with colliding naming schemes.  To keep with the abstraction that currently exists in the Mixed modulestore, we should handle the case for duplicate course_ids across modulestores.  And giving precedence to the first-found course_id makes sense assuming the list of the modulestores is intentionally ordered.